src/app/services/tasksApi.ts

```diff
@@ -5,30 +5,58 @@ import task, {
 } from '@/interfaces/task.type';
 import { api } from './api';
 import { MINE_TASKS_URL, TASKS_URL } from '@/constants/url';
+import { ASSIGNED } from '@/constants/task-status';
+import {
+    THOUSAND_MILLI_SECONDS,
+    FOURTEEN_DAYS,
+    SECONDS_IN_A_DAY,
+} from '@/constants/date';
 
 type TasksCreateMutationResponse = { message: string; task: task };
+type AssignTaskPayload = { taskId: string; assignee: string };
+
+export const assignTaskReducerStateBuilder = () => {
+    const dateObject: Date = new Date();
+    const startedOnEpoch: number =
+        dateObject.getTime() / THOUSAND_MILLI_SECONDS;
+    const endsOnEpoch: number =
+        startedOnEpoch + FOURTEEN_DAYS * SECONDS_IN_A_DAY;
+    const build = ({ assignee }: { assignee: string }) => {
+        return {
+            status: ASSIGNED,
+            startedOn: startedOnEpoch,
+            endsOn: endsOnEpoch,
+            assignee,
+        };
+    };
+    return build;
+};
 
 export const tasksApi = api.injectEndpoints({
     endpoints: (builder) => ({
         getAllTasks: builder.query<TasksResponseType, GetAllTaskParamType>({
             query: ({ dev, status, nextPage, prevPage }) => {
-                let url = dev ? `/tasks?status=${status}&dev=true` : '/tasks';
-
-                if (nextPage) {
-                    url = nextPage;
-                }
-
-                if (prevPage) {
-                    url = prevPage;
-                }
+                const url =
+                    nextPage ?? prevPage ?? dev
+                        ? `/tasks?status=${status}&dev=true`
+                        : '/tasks';
 
                 return { url };
             },
-            providesTags: ['Tasks'],
+            providesTags: (result) =>
+                result
+                    ? [
+                          ...result['tasks'].map(({ id }) => ({
+                              type: 'Tasks' as const,
+                              id,
+                          })),
+                          'Tasks',
+                      ]
+                    : ['Tasks'],
 
             transformResponse: (response: TasksResponseType) => {
                 return {
-                    tasks: response.tasks?.sort(
+                    tasks: [...response.tasks].sort(
```

Comment:
why made this change?

Reply:
> why made this change?
> 

The previous code might mutate the response, so i created a new array and pass the required changes without mutating the response